tests/README.md

```diff
@@ -0,0 +1,120 @@
+# Honey-badger tests
+
+The `hbbft` crate comes with a toolkit for testing its various algorithms in simulated network environments.
+
+## Old vs new
+
+The old testing code can be found inside the `network` module and `.rs` files in the `tests` subdirectory that are not prefixed with `net_`. The newer networking code is contained inside the `net` module and the remaining `.rs` files.
+
+## VirtualNet
+
+Core of most tests is the `net::VirtualNet` struct, which simulates a network of nodes all running an instance of a distributed algorithm. Messages sent by these nodes are queued by the network. Every time the network is "cranked", a buffered message is delivered to its destination node and processed.
+
+Virtual networks can also host an adversary that can affect faulty nodes (which are tracked automatically) or reorder queued messages.
+
+To create a new network, the `NetBuilder` should be used:
+
+```rust
+// Create a network of 10 nodes, out of which 3 are faulty.
+let mut net = NetBuilder::new(0..10)
+    .num_faulty(3)
+    .using(move |node| { DynamicHoneyBadger::builder().build(node.netinfo) })
+    .build()
+    .expect("could not construct test network");
+```
+
+Algorithms that return a `Step` upon construction should use `using_step` instead.
```

Comment:
Can you clarify the algorithms that require using_step and provide a 1 line example?

Reply:
Will add one, albeit later.